src/publisher/tests/test_utils.py

```diff
@@ -131,3 +131,21 @@ def test_todict(self):
         jnl_data = utils.to_dict(jnl)
         self.assertTrue(isinstance(jnl_data, dict))
         self.assertEqual(jnl_data['name'], settings.PRIMARY_JOURNAL['name'])
+
+    def test_has_all_keys(self):
+        data = {'a': 1, 'b': 2}
+        cases = [
+            (data, ['a'], True),
+            (data, ['a', 'b'], True),
+            (data, ['a', 'b', 'c'], False),
+            (data, [0], False),
+            (data, [0, 1], False),
+            (data, [self], False),
+        ]
+        for case, args, expected in cases:
+            try:
+                self.assertEqual(utils.has_all_keys(case, args), expected)
```

Comment:
ha. I used to know this but forgot. good call. I guess it's just wrapping
the:

assert cond, msg

in python already

On Wed, Jun 29, 2016 at 8:15 AM, Giorgio Sironi <notifications@github.com>
wrote:

> In src/publisher/tests/test_utils.py
> <https://github.com/elifesciences/lax/pull/18#discussion_r68895994>:
>
> > @@ -131,3 +131,21 @@ def test_todict(self):
> >          jnl_data = utils.to_dict(jnl)
> >          self.assertTrue(isinstance(jnl_data, dict))
> >          self.assertEqual(jnl_data['name'], settings.PRIMARY_JOURNAL['name'])
> > +
> > +    def test_has_all_keys(self):
> > +        data = {'a': 1, 'b': 2}
> > +        cases = [
> > +            (data, ['a'], True),
> > +            (data, ['a', 'b'], True),
> > +            (data, ['a', 'b', 'c'], False),
> > +            (data, [0], False),
> > +            (data, [0, 1], False),
> > +            (data, [self], False),
> > +        ]
> > +        for case, args, expected in cases:
> > +            try:
> > +                self.assertEqual(utils.has_all_keys(case, args), expected)
>
> a passing comment: you can do
>
> self.assertEqual(
>     utils.has_all_keys(case, args),
>     expected,
>     "case: %s; args: %s" % (case, args)
> )
>
> to avoid the try/except/raise
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/elifesciences/lax/pull/18/files/2f760b32cbff80627b93faeb6485a9a0bd0e2e09#r68895994>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACCyHFk8MZUFd_DSjjaINatwgd0AW04nks5qQhupgaJpZM4JAUql>
> .
>



-- 

Luke Skibinski

Formally defined,
Extensively tested,
Cryptic, abstract, thing.

--

eLife Sciences Publications, Ltd is a limited liability non-profit
non-stock corporation incorporated in the State of Delaware, USA, with
company number 5030732, and is registered in the UK with company number
FC030576 and branch number BR015634 at the address First Floor, 24 Hills
Road, Cambridge CB2 1JP.